packages/cli/src/lib/new/preparation/loadPortableTemplateConfig.ts

```diff
@@ -0,0 +1,159 @@
+import { paths } from '../../paths';
+import { defaultTemplates } from '../defaultTemplates';
+import {
+  PortableTemplateConfig,
+  PortableTemplatePointer,
+  TEMPLATE_FILE_NAME,
+} from '../types';
+import { parse as parseYaml } from 'yaml';
+import { z } from 'zod';
+import { fromZodError } from 'zod-validation-error';
+import { ForwardedError } from '@backstage/errors';
+
+const defaults = {
+  license: 'Apache-2.0',
+  version: '0.1.0',
+  private: true,
+  publishRegistry: undefined,
+  packageNamePrefix: '@internal/',
+  packageNamePluginInfix: 'plugin-',
+};
+
+const newConfigSchema = z
+  .object({
+    templates: z.array(z.string()).optional(),
+    globals: z
+      .object({
+        license: z.string().optional(),
+        version: z.string().optional(),
+        private: z.boolean().optional(),
+        publishRegistry: z.string().optional(),
+        namePrefix: z.string().optional(),
+        namePluginInfix: z.string().optional(),
+      })
+      .optional(),
+  })
+  .strict();
+
+const pkgJsonWithNewConfigSchema = z.object({
+  backstage: z
+    .object({
+      cli: z
+        .object({
+          new: newConfigSchema.optional(),
+        })
+        .optional(),
+    })
+    .optional(),
+});
+
+type LoadConfigOptions = {
+  packagePath?: string;
+  overrides?: Partial<PortableTemplateConfig>;
+};
+
+export async function loadPortableTemplateConfig(
+  options: LoadConfigOptions = {},
+): Promise<PortableTemplateConfig> {
+  const { overrides = {} } = options;
+  const pkgPath =
+    options.packagePath ?? paths.resolveTargetRoot('package.json');
+  const pkgJson = await fs.readJson(pkgPath);
+
+  const parsed = pkgJsonWithNewConfigSchema.safeParse(pkgJson);
+  if (!parsed.success) {
+    throw new ForwardedError(
+      `Failed to load templating configuration from '${pkgPath}'`,
+      fromZodError(parsed.error),
+    );
+  }
+
+  const config = parsed.data.backstage?.cli?.new;
+
+  const basePath = dirname(pkgPath);
+  const templatePointers = await Promise.all(
+    (config?.templates ?? defaultTemplates).map(async pointer => {
+      try {
+        const templatePath = resolveLocalTemplatePath(pointer, basePath);
+
+        return await peekLocalTemplateDefinition(templatePath);
+      } catch (error) {
+        throw new ForwardedError(
+          `Failed to load template definition '${pointer}'`,
+          error,
+        );
+      }
+    }),
+  );
+
+  return {
+    isUsingDefaultTemplates: !config?.templates,
+    templatePointers,
+    license: overrides.license ?? config?.globals?.license ?? defaults.license,
+    version: overrides.version ?? config?.globals?.version ?? defaults.version,
+    private: overrides.private ?? config?.globals?.private ?? defaults.private,
+    publishRegistry:
+      overrides.publishRegistry ??
+      config?.globals?.publishRegistry ??
+      defaults.publishRegistry,
+    packageNamePrefix:
+      overrides.packageNamePrefix ??
+      config?.globals?.namePrefix ??
```

Comment:
any particular reason that the config globals are differently named? (here and below)

Reply:
I added the `package*` prefix to all internal representation for extra context. Had them wrapped in a separate object for a while too but felt it was a bit nicer to flatten things out in the end. On the other hand I think `packageNamePrefix` etc. is a bit too verbose for config